metaflow/plugins/kfp/kfp.py

```diff
@@ -404,19 +418,48 @@ def step_op_func(
     print("----")
 
     # TODO: Map username to KFP specific user/profile/namespace
+    # Note: we don't put this in a try catch block as below because
+    # Popen will not produce a error that will cause the Kubeflow step
+    # to stop
     with Popen(
         ["/bin/sh", "-c", cmd],
         stdin=PIPE,
         stdout=PIPE,
         stderr=STDOUT,
         universal_newlines=True,
         env=dict(os.environ, USERNAME="kfp-user", METAFLOW_RUN_ID=kfp_run_id),
-    ) as process, StringIO() as string_buffer:
+    ) as process, StringIO() as stdout_buffer:
         for line in process.stdout:
             print(line, end="")
-            string_buffer.write(line)
-        buffer_output = string_buffer.getvalue()
-    print("___ DONE ___")
+            stdout_buffer.write(line)
+        stdout_output = stdout_buffer.getvalue()
+
+    with open("0.stdout.log", "w") as stdout_file, open(
+        "0.stderr.log", "w"
+    ) as stderr_file:
+        _ = stdout_file.write(stdout_output)
+
+    # TODO: obtain the string `s3://kfp-example-aip-dev/metaflow/` dynamically
+    save_logs_cmd_template = (
+        f"python -m awscli s3 cp {{log_file}} {datastore_root}/"
+        f"{flow_name}/{kfp_run_id}/{step_name}/"
+        f"{task_id}/{{log_file}}"
+    )
```

Comment:
I know that I previously asked to use the Metaflow S3 client.
However can we instead update the bash command to be executed (see https://github.com/zillow/metaflow/blob/1b861af0b9c6f81522b6e53ebf6ca88860ef5f1f/metaflow/plugins/kfp/kfp.py#L320), to pipe stdout to a log file, then at the end run the `python -m awscli s3 cp ` command?

Hence `step_op_func` wouldn't change.

Reply:
https://stackoverflow.com/questions/7901517/how-to-redirect-stderr-and-stdout-to-different-files-in-the-same-line-in-script
https://stackoverflow.com/questions/637827/redirect-stderr-and-stdout-in-bash

and we can use `tee` to print it to the console for KFP UX output.  That'd remove the need for Python code `stdout_buffer` and `stderr_buffer`, and the for loop to print it, and writing the log files.